File: Backend/src/main/java/com/example/Inventra/service/AlertService.java
package com.example.Inventra.service;

import com.example.Inventra.dto.StockAlertResponseDTO;
import com.example.Inventra.entity.Product;

import java.util.List;

public interface AlertService {
    void checkAndFireAlert(Product product);
    void resolveAlert(Long productId, String resolvedBy);
    List<StockAlertResponseDTO> getOpenAlerts();
    List<StockAlertResponseDTO> getAlertsByProduct(Long productId);
}

File: frontend/inventra-frontend/src/services/alertService.js
import api from './api';

export const getOpenAlerts = () => api.get('/alerts');

export const getAlertsByProduct = (productId) =>
  api.get(`/alerts/product/${productId}`);

export const resolveAlert = (productId, resolvedBy) =>
  api.patch(`/alerts/resolve/${productId}?resolvedBy=${encodeURIComponent(resolvedBy)}`);